src/main/java/com/example/macro/user/service/UserService.java

```diff
@@ -0,0 +1,82 @@
+package com.example.macro.user.service;
+
+import com.example.macro.user.document.User;
+import com.example.macro.user.exception.DuplicateUserException;
+import com.example.macro.user.exception.UserNotFoundException;
+import com.example.macro.user.repository.UserRepository;
+import com.example.macro.service.GenericService;
+import com.example.macro.user.utils.EmailValidation;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class UserService extends GenericService<User, String> {
+
+    private final BCryptPasswordEncoder passwordEncoder;
+
+    private final UserRepository userRepository;
+
+
+
+    public UserService(BCryptPasswordEncoder passwordEncoder, UserRepository userRepository) {
+        this.passwordEncoder = passwordEncoder;
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    public User save(User user) {
+        if (existingUser(user.getEmail())) {
+            throw new DuplicateUserException("Record with email " + user.getUserName() + " Already Exists");
+        }
+
+        if (EmailValidation.isEmailValid(user.getEmail())) {
+            user.setPassword(passwordEncoder.encode(user.getPassword()));
+            return userRepository.save(user);
+        }
+        return new User();
```

Comment:
I really think we need a better option for a failed save than returning an empty user object